src/openms/source/ANALYSIS/ID/FragmentIndex.cpp

```diff
@@ -119,64 +116,85 @@ namespace OpenMS
   }
 
   void FragmentIndex::build(const std::vector<FASTAFile::FASTAEntry>& fasta_entries)
   {
       // get the spectrum generator and set the ion-types
       TheoreticalSpectrumGenerator tsg;
       auto tsg_params = tsg.getParameters();
-      tsg_params.setValue("add_a_ions", add_a_ions_);
-      tsg_params.setValue("add_b_ions", add_b_ions_);
-      tsg_params.setValue("add_c_ions", add_c_ions_);
-      tsg_params.setValue("add_x_ions", add_x_ions_);
-      tsg_params.setValue("add_y_ions", add_y_ions_);
-      tsg_params.setValue("add_z_ions", add_z_ions_);
-      tsg.setParameters(tsg_params); // TODO: this was missing before!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+      auto this_params = getParameters();
+      tsg_params.setValue("add_a_ions", this_params.getValue("add_a_ions"));
+      tsg_params.setValue("add_b_ions", this_params.getValue("add_b_ions"));
+      tsg_params.setValue("add_c_ions", this_params.getValue("add_c_ions"));
+      tsg_params.setValue("add_x_ions", this_params.getValue("add_x_ions"));
+      tsg_params.setValue("add_y_ions", this_params.getValue("add_y_ions"));
+      tsg_params.setValue("add_z_ions", this_params.getValue("add_z_ions"));
+      tsg.setParameters(tsg_params);
       PeakSpectrum b_y_ions;
 
       /// generate all Peptides
       generate_peptides(fasta_entries);
 
-      size_t peptide_idx = 0;
+      cout << "test2" <<endl;
+      /// Since we (the new) Peptide struct does not store the AASequence, we must reconstruct the modified ones
+      /// therefore we need the modificationGenerators:
+      ModifiedPeptideGenerator::MapToResidueType fixed_modifications = ModifiedPeptideGenerator::getModifications(modifications_fixed_);
+      ModifiedPeptideGenerator::MapToResidueType variable_modifications = ModifiedPeptideGenerator::getModifications(modifications_variable_);
+
+      UInt32 peptide_idx = 0;
+
 
-      /// For each Peptides get all theoretical b and y ions // TODO: include other fragmentation methods
-      for (const Peptide& pep: fi_peptides_)
+      for (const sPeptide& pep: fi_peptides_)
       {
-        tsg.getSpectrum(b_y_ions, pep.sequence, 1, 1);
+        auto debugingStuff = fasta_entries[pep.protein_idx_].sequence;
+        AASequence unmod_peptide = AASequence::fromString(fasta_entries[pep.protein_idx_].sequence.substr(pep.sequence.first, pep.sequence.second));
+        vector<AASequence> mod_peptides;
+        if(!(modifications_fixed_.empty() && modifications_variable_.empty()))
+        {
+          AASequence mod_peptide = AASequence(unmod_peptide); // copy the peptide
+          ModifiedPeptideGenerator::applyFixedModifications(fixed_modifications, mod_peptide);
+          ModifiedPeptideGenerator::applyVariableModifications(variable_modifications, mod_peptide, max_variable_mods_per_peptide_, mod_peptides);
+          tsg.getSpectrum(b_y_ions, mod_peptides[pep.modification_idx], 1,1);
```

Comment:
Note: modifications that fragment (e.g. TMT) need to be removed before generating the spectrum!

Reply:
Is there a fast way to tell if a modification is removed during Fragmentation?

Reply:
not really. I think we could add some special option/extra flag to the search engine to support TMT